httptesting/http.go

```diff
@@ -321,3 +321,36 @@ type readSeekNopCloser struct {
 func (readSeekNopCloser) Close() error {
 	return nil
 }
+
+// Transport is an http.RoundTripper that can rewrite request URLs. If
+// the request URL has the prefix specified in Match that part will be
+// changed to the value specified in Replace. RoundTripper will then be
+// used to perform the resulting request. If RoundTripper is nil
+// http.DefaultTransport will be used.
```

Comment:
URLRewritingTransport?

Also, perhaps mention the usual use case?

//
// This can be used in tests that, for whatever reason,
// need to make a call to a URL that's not in our control
// but we want to control the results of HTTP requests
// to that URL.